### ISSUE TITLE
Request support about the flash attention 2 for the gpt_bigcode models

### ISSUE THREAD

**Issue by DongHande** [huggingface/transformers#26857]
When I run the code,  
```
checkpoint = "bigcode/starcoder"
model = AutoModelForCausalLM.from_pretrained(checkpoint, use_flash_attention_2=True)
```
it shows that "ValueError: The current architecture does not support Flash Attention 2.0. Please open an issue on GitHub to request support for this architecture: https://github.com/huggingface/transformers/issues/new". 

The reason is that the gpt_bigcode (https://github.com/huggingface/transformers/blob/main/src/transformers/models/gpt_bigcode/modeling_gpt_bigcode.py) does not support flash attention 2 yet. 
I open this issue to request support about the flash attention 2 for the gpt_bigcode models.

**Comment by ArthurZucker** [comment]
Hey! Will be fixed by #26479

**Comment by DongHande** [comment]
Hope it will be soon fixed and merged.
